std/haxe/macro/ExprTools.hx

```diff
@@ -130,7 +130,8 @@ class ExprTools {
 					f(edef);
 			case EFormat(parts):
 				for (part in parts) switch part.kind {
-					case FRaw(_) | FIdent(_):
+					case FRaw(_):
+					case FIdent(i): f({pos: part.pos, expr: EConst(CIdent(i))});
```

Comment:
I think I'll just keep the original position FIdent position as back2dos originally proposed.